src/main/java/com/cisco/trex/stateful/TRexAstfClient.java

```diff
@@ -368,4 +372,92 @@ public String getVersion() {
         }
     }
 
+    /**
+     * get template group names
+     *
+     * @return template group names
+     */
+    public List<String> getTemplateGroupNames() {
+        return this.getTemplateGroupNames(null);
+    }
+
+    /**
+     * get template group names
+     *
+     * @param profileId
+     * @return template group names
+     */
+    public List<String> getTemplateGroupNames(String profileId) {
+        Map<String, Object> payload = createPayload(profileId);
+        payload.put("epoch", 1);
+        payload.put("initialized", false);
+        String json = callMethod("get_tg_names", payload);
+        JsonElement response = new JsonParser().parse(json);
+        JsonArray names = response.getAsJsonArray().get(0).getAsJsonObject().get("result").getAsJsonObject()
+                .get("tg_names").getAsJsonArray();
+        return StreamSupport.stream(names.spliterator(), false).map(name -> name.getAsString())
+                .collect(Collectors.toList());
+
+    }
+
+    /**
+     * get template group statistics
+     *
+     * @param tgNames
+     * @return
+     */
+    public Map<String, AstfStatistics> getTemplateGroupStatistics(List<String> tgNames) {
+        return getTemplateGroupStatistics(null, tgNames);
+    }
+
+    /**
+     * get template group statistics
+     *
+     * @param profileId
+     * @param tgNames
+     * @return
+     */
+    public Map<String, AstfStatistics> getTemplateGroupStatistics(String profileId, List<String> tgNames) {
+        Map<String, AstfStatistics> stats = new HashMap<>();
+        Map<String, Object> payload = createPayload(profileId);
+        payload.put("epoch", 1);
+
+        Map<String, Integer> name2Id = translateNames2Ids(profileId, tgNames);
+        List<Integer> tgIds = new ArrayList<>(tgNames.size());
+        tgNames.forEach(name -> tgIds.add(name2Id.get(name)));
+        payload.put("tg_ids", tgIds);
+
+        String json = callMethod("get_tg_id_stats", payload);
+        JsonElement response = new JsonParser().parse(json);
+
+        JsonObject result = response.getAsJsonArray().get(0).getAsJsonObject().get("result").getAsJsonObject();
+        Map<Integer, String> id2Name = translateIds2Names(name2Id);
+        for (Integer id : tgIds){
+            stats.put(id2Name.get(id), GSON.fromJson(result.get(id.toString()), AstfStatistics.class));
```

Comment:
use jackson-databind APIs to do the java serialization:
new ObjectMapper().readValue(result.get(id.toString(), AstfStatistics.class).
then you don't need change clientCounters/serverCounters names in AstfStatistics.java.

 AstfStatistics is following the jackson format to define the json fields (using @JsonProperty tag), it only works in case java properties have the same name with json fields when using GSON to parse it,.